### PR TITLE
Rest api to search for requests

### PR DESCRIPTION
**Pull request by sdrafahl**
## Rest API To Search For Requests

https://trello.com/c/4GuxHxor/32-rest-api-to-search-for-requests

## Short Descriptions

Create Restful API on the express side to allow for searching of requests by nearest location given a address and a max number of requests wanted in JSON form.

### CODE REVIEW

File: myRental/modules/database.js [hunk 1]
```diff
@@ -103,37 +110,45 @@ method.handleRequest = (request, callBack) => {
...
-            .then(() => {
-                return callBack ({
-                    success: true,
-                });
-            })
-            .catch((error) => {
-                logger.log(error);
-                return callBack ({
-                    success: false,
+        let addressString = address + " " + city + " , " + state + " " + zip + " " + country;
```
Comment (jd1388) [inline myRental/modules/database.js]: You can try doing ```let addressString = `${address} ${city} ${state} ${zip} ${country}`; ``` if this makes more sense.

File: myRental/test/test.js [hunk 2]
```diff
@@ -88,58 +84,27 @@ describe('requestController', () => {
...
+            let testUserRequest = helperFunctions.generateFakeUserRequest();
+            let serviceRequest1 = helperFunctions.generateFakeServiceRequest1();
+            let serviceRequest2 = helperFunctions.generateFakeServiceRequest2();
+            let serviceRequest3 = helperFunctions.generateFakeServiceRequest3();
+            database.registerUser(testUserRequest, (cb) => {
+                database.handleRequest(serviceRequest1, (cb) => {
+                    database.handleRequest(serviceRequest2, (cb) => {
+                        database.handleRequest(serviceRequest3, (cb) => {
+                            database.searchForRequests(helperFunctions.generateSearchRequest(), (models) => {
+                                assert.equal(models.data[0].streetAddress, '201 E 17th St N');
```
Comment (jd1388) [inline myRental/test/test.js]: Is there a way to use a random address?

File: myRental/test/testRequests.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
+let faker = require('faker');
+let crypto = require('crypto');
+let config = require('../config.json');
+
+function generateSearchRequest() {
+    return {
+        'body': {
+            'max': 3,
+            'address': '800-820 E 15th St N',
```
Comment (jd1388) [inline myRental/test/testRequests.js]: Is there a way to use faker to generate these addresses? Does it even need to look like an address for testing purposes?

File: myRental/test/test.js (hunk 2 above)
Comment (sdrafahl) [reply myRental/test/test.js]: I dont think making it random is practical.

File: myRental/test/testRequests.js (hunk 3 above)
Comment (sdrafahl) [reply myRental/test/testRequests.js]: Same as comment above.

**Pull request merged**